### PR TITLE
Rebase master gui scene name

### PR DESCRIPTION
**Pull request by AdrianHarenczyk**
- Add new scene with name of player
- Change other scenes according to new data which is name of current player
- Added connection choosing box which has for now one option
- Refactor of packages, now it's more clear what is where

**Please check for any cleancode violations.
Maven clean install results in success.**

### CODE REVIEW

File: client/src/main/java/gui/controllers/PlacementSceneController.java [hunk 1]
```diff
@@ -61,8 +74,18 @@ private void startTheGame() throws Exception {
...
             Stage currentStage = (Stage) currentWindow;
-            GameStage gameStage = GameStage.createGameStage();
-            gameStage.start(currentStage);
+            GameScene gameScene = GameScene.create();
+            gameScene.start(currentStage);
+        }
+    }
+
+    @FXML
+    private void backToNameChoosing() throws Exception {
```
Comment (KrzysztofDzioba) [inline client/src/main/java/gui/controllers/PlacementSceneController.java]: Can throw IOException

**Review by KrzysztofDzioba — APPROVED** [review]
Looks OK

File: client/src/main/java/gui/controllers/GameSceneController.java [hunk 2]
```diff
@@ -23,6 +31,8 @@
 
     public void initialize() {
         appInstance = ClientAppRunner.getInstance();
+        String playerName = appInstance.getPlayer().getName();
```
Comment (makoJohn) [inline client/src/main/java/gui/controllers/GameSceneController.java:34]: you can inline this variable

File: client/src/main/java/gui/data/Player.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package gui.data;
+
+import java.util.Objects;
+
+public class Player {
```
Comment (makoJohn) [inline client/src/main/java/gui/data/Player.java:5]: good candidate for AutoValue :)

File: client/src/main/java/gui/data/ServerInfo.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
+package gui.data;
+
+import java.util.Objects;
+
+public class ServerInfo {
+    private String host;
```
Comment (makoJohn) [inline client/src/main/java/gui/data/ServerInfo.java]: final

File: client/src/main/java/gui/instance/ClientAppRunner.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+ * This is the singleton class. It's purpose is to
+ * provide information flow between controller instances.
+ */
+public class ClientAppRunner {
+
+    private FleetView fleet;
+    private ServerConnection serverConnection;
+    private Player currentPlayer;
+
+    private static ClientAppRunner ourInstance = new ClientAppRunner();
```
Comment (makoJohn) [inline client/src/main/java/gui/instance/ClientAppRunner.java:18]: why singleton?

File: client/src/main/java/gui/controllers/PlacementSceneController.java [hunk 6]
```diff
@@ -108,12 +131,7 @@ private Runnable addToTextArea(String message) {
     }
 
     private void disableStartWhenNoFleet() {
-        boolean fleetDoesNotExist = (fleet == null);
-        if (fleetDoesNotExist) {
-            startButton.setDisable(true);
-        } else {
-            startButton.setDisable(false);
-        }
+        startButton.setDisable(fleet == null);
```
Comment (boczkas) [inline client/src/main/java/gui/controllers/PlacementSceneController.java]: maybe extract boolean?

File: client/src/main/java/gui/controllers/PlayerSceneController.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+import javafx.stage.Stage;
+import javafx.stage.Window;
+
+public class PlayerSceneController {
+    @FXML
+    private Button startPlacement;
+    private ClientAppRunner appInstance;
+    private ObservableList<ServerInfo> listOfServers;
+    private static final int PORT_NUMBER = 7777;
+    @FXML
```
Comment (boczkas) [inline client/src/main/java/gui/controllers/PlayerSceneController.java]: \n

File: client/src/main/java/gui/controllers/PlayerSceneController.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    private ComboBox<ServerInfo> serverNameComboBox;
+    @FXML
```
Comment (boczkas) [inline client/src/main/java/gui/controllers/PlayerSceneController.java]: \n

File: client/src/main/java/gui/controllers/PlayerSceneController.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    private TextField playerNameField;
+
+    public void initialize() {
+        appInstance = ClientAppRunner.getInstance();
+        listOfServers = FXCollections.observableArrayList(ServerInfo.create("localhost", PORT_NUMBER));
+        serverNameComboBox.setItems(listOfServers);
+        serverNameComboBox.getSelectionModel().select(0);
+    }
+
```
Comment (boczkas) [inline client/src/main/java/gui/controllers/PlayerSceneController.java:35]: remove \n

File: client/src/main/java/gui/instance/ClientAppRunner.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    public Player getPlayer() {
+        return currentPlayer;
+    }
+
+    public ServerConnection getServerConnection() {
+        return serverConnection;
+    }
+
+
```
Comment (boczkas) [inline client/src/main/java/gui/instance/ClientAppRunner.java]: remove new line

File: client/src/main/java/gui/instance/ClientAppRunner.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    public void initializeServerConnection(ServerInfo serverInfo) {
+        if (checkIfDisconnected()) {
+            connectToServer(serverInfo);
+        }
+    }
+
+    private boolean checkIfDisconnected() {
+        return null == serverConnection;
```
Comment (boczkas) [inline client/src/main/java/gui/instance/ClientAppRunner.java:52]: consider different options than null checking

File: client/src/main/java/gui/instance/ClientAppRunner.java [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    private void connectToServer(ServerInfo serverInfo) {
+        int port = serverInfo.getPort();
+        String host = serverInfo.getHost();
+        serverConnection = ServerConnection.initializeConnection(host, port);
+        serverConnection.createServerConnection();
+    }
+
```
Comment (boczkas) [inline client/src/main/java/gui/instance/ClientAppRunner.java]: remove new lines

File: client/src/main/java/gui/scenes/PlayerScene.java [hunk 13]
```diff
@@ -7,20 +7,23 @@
...
+        Parent root = FXMLLoader.load(getClass().getResource("/player_scene.fxml"));
         primaryStage.setTitle("Battleships OStatki");
         primaryStage.setScene(new Scene(root, SCENE_WIDTH, SCENE_HEIGHT));
         primaryStage.resizableProperty().setValue(false);
         primaryStage.show();
     }
-    static MainStage createMainStage() {
-        return new MainStage();
-    }
+
```
Comment (boczkas) [inline client/src/main/java/gui/scenes/PlayerScene.java]: remove new line

**Review by boczkas — CHANGES_REQUESTED** [review]
check comments

File: client/src/main/java/gui/controllers/GameSceneController.java (hunk 2 above)
Comment (AdrianHarenczyk) [reply client/src/main/java/gui/controllers/GameSceneController.java:34]: I think it's more clear that way. That was my intention.

File: client/src/main/java/gui/data/Player.java (hunk 3 above)
Comment (AdrianHarenczyk) [reply client/src/main/java/gui/data/Player.java:5]: Thank you for suggestion. We don't use for now AutoValue, maybe in the next project. :)

File: client/src/main/java/gui/instance/ClientAppRunner.java (hunk 5 above)
Comment (AdrianHarenczyk) [reply client/src/main/java/gui/instance/ClientAppRunner.java:18]: To save some informations through the game scenes.

File: client/src/main/java/gui/data/ServerInfo.java (hunk 4 above)
Comment (AdrianHarenczyk) [reply client/src/main/java/gui/data/ServerInfo.java]: Done.

File: client/src/main/java/gui/controllers/PlacementSceneController.java (hunk 6 above)
Comment (AdrianHarenczyk) [reply client/src/main/java/gui/controllers/PlacementSceneController.java]: Done

File: client/src/main/java/gui/controllers/PlayerSceneController.java (hunk 7 above)
Comment (AdrianHarenczyk) [reply client/src/main/java/gui/controllers/PlayerSceneController.java]: Done

File: client/src/main/java/gui/controllers/PlayerSceneController.java (hunk 8 above)
Comment (AdrianHarenczyk) [reply client/src/main/java/gui/controllers/PlayerSceneController.java]: Done

File: client/src/main/java/gui/controllers/PlayerSceneController.java (hunk 9 above)
Comment (AdrianHarenczyk) [reply client/src/main/java/gui/controllers/PlayerSceneController.java:35]: Done

File: client/src/main/java/gui/instance/ClientAppRunner.java (hunk 10 above)
Comment (AdrianHarenczyk) [reply client/src/main/java/gui/instance/ClientAppRunner.java]: Done

File: client/src/main/java/gui/instance/ClientAppRunner.java (hunk 12 above)
Comment (AdrianHarenczyk) [reply client/src/main/java/gui/instance/ClientAppRunner.java]: Done

File: client/src/main/java/gui/scenes/PlayerScene.java (hunk 13 above)
Comment (AdrianHarenczyk) [reply client/src/main/java/gui/scenes/PlayerScene.java]: Done

File: client/src/main/java/gui/instance/ClientAppRunner.java (hunk 11 above)
Comment (AdrianHarenczyk) [reply client/src/main/java/gui/instance/ClientAppRunner.java:52]: We will replace it further down the road. Issue is tracked under number #29.

**Review by boczkas — APPROVED** [review]
:+1:

**Pull request merged**